src/test/modelconfig_test.cpp

```diff
@@ -656,6 +656,198 @@ TEST(ModelConfig, ConfigParseNodeWithValidShapeFormatArray) {
+}
+
+TEST(ModelConfig, ConfigParseCacheDisabledForBatchAutoByDefault) {
+    std::string config = R"#(
+        {"model_config_list": [{
+            "config": {
+                "name": "alpha",
+                "base_path": "/tmp/models/alpha",
+                "batch_size": "auto"
+            }
+        }]}
+    )#";
+
+    rapidjson::Document configJson;
+    rapidjson::ParseResult parsingSucceeded = configJson.Parse(config.c_str());
+    ASSERT_EQ(parsingSucceeded, true);
+
+    const auto modelConfigList = configJson.FindMember("model_config_list");
+    ASSERT_NE(modelConfigList, configJson.MemberEnd());
+    const auto& configs = modelConfigList->value.GetArray();
+    ASSERT_EQ(configs.Size(), 1);
+    ovms::ModelConfig modelConfig;
+    auto status = modelConfig.parseNode(configs[0]["config"]);
+
+    ASSERT_EQ(status, ovms::StatusCode::OK);
+    EXPECT_TRUE(modelConfig.isCachingDisabled());
+}
+
+TEST(ModelConfig, ConfigParseCacheDisabledForShapeAutoByDefault) {
+    std::string config = R"#(
+        {"model_config_list": [{
+            "config": {
+                "name": "alpha",
+                "base_path": "/tmp/models/alpha",
+                "shape": "auto"
+            }
+        }]}
+    )#";
+
+    rapidjson::Document configJson;
+    rapidjson::ParseResult parsingSucceeded = configJson.Parse(config.c_str());
+    ASSERT_EQ(parsingSucceeded, true);
+
+    const auto modelConfigList = configJson.FindMember("model_config_list");
+    ASSERT_NE(modelConfigList, configJson.MemberEnd());
+    const auto& configs = modelConfigList->value.GetArray();
+    ASSERT_EQ(configs.Size(), 1);
+    ovms::ModelConfig modelConfig;
+    auto status = modelConfig.parseNode(configs[0]["config"]);
+
+    ASSERT_EQ(status, ovms::StatusCode::OK);
+    EXPECT_TRUE(modelConfig.isCachingDisabled());
+}
+
+TEST(ModelConfig, ConfigParseCacheCannotForceEnableForCustomLoaders) {
+    std::string config = R"#(
+        {"model_config_list": [{
+            "config": {
+                "name": "alpha",
+                "base_path": "/tmp/models/alpha",
+                "custom_loader_options": {
+                    "loader_name": "sampleloader",
+                    "model_file": "resnet50-binary-0001.xml",
+                    "bin_file": "resnet50-binary-0001.bin"
+                },
+                "allow_cache": true
+            }
+        }]}
+    )#";
+
+    rapidjson::Document configJson;
+    rapidjson::ParseResult parsingSucceeded = configJson.Parse(config.c_str());
+    ASSERT_EQ(parsingSucceeded, true);
+
+    const auto modelConfigList = configJson.FindMember("model_config_list");
+    ASSERT_NE(modelConfigList, configJson.MemberEnd());
+    const auto& configs = modelConfigList->value.GetArray();
+    ASSERT_EQ(configs.Size(), 1);
+    ovms::ModelConfig modelConfig;
+    auto status = modelConfig.parseNode(configs[0]["config"]);
+
+    ASSERT_EQ(status, ovms::StatusCode::OK);
+    EXPECT_TRUE(modelConfig.isCachingDisabled());
```

Comment:
I think that we shouldn't at all load the model when the configuration is wrong and at least log the error.

Reply:
We log model cache setting in debug mode:
https://github.com/openvinotoolkit/model_server/blob/develop/src/modelinstance.cpp#L508
@atobiszei is it enough?

Reply:
After sync we agreed this is not enough. Agreed also to merge this and treat this as bug, ticket: https://jira.devtools.intel.com/browse/CVS-71598